stan/math/prim/fun/log_mix.hpp

```diff
@@ -157,38 +91,19 @@ return_type_t<T_theta, T_lam> log_mix(const T_theta& theta,
   check_finite(function, "theta", theta);
   check_consistent_sizes(function, "theta", theta, "lambda", lambda);
 
-  scalar_seq_view<T_theta> theta_vec(theta);
-  T_partials_vec theta_dbl(N);
-  for (int n = 0; n < N; ++n) {
-    theta_dbl[n] = value_of(theta_vec[n]);
-  }
-
-  scalar_seq_view<T_lam> lam_vec(lambda);
-  T_partials_vec lam_dbl(N);
-  for (int n = 0; n < N; ++n) {
-    lam_dbl[n] = value_of(lam_vec[n]);
-  }
+  const auto& theta_dbl = value_of(as_column_vector_or_scalar(theta));
```

Comment:
```suggestion
  decltype(auto) theta_dbl = value_of(as_column_vector_or_scalar(theta));
```
Could be more succinct

Reply:
What you are suggesting is actually longer.